### PR TITLE
(WIP) Refactor to es modules

### PR DESCRIPTION
**Pull request by peterschussheim**
Associated Issue: #1884 

## Summary of Changes

(WIP) Refactor the following files to use ES module syntax:
- utils
	- editor
		- [x] build-query.js
		- [x] expression.js
		- [ ] index.js
		- [x] source-documents.js
		- [x] source-search.js
	- redux
		- middleware
			- [x] history.js
			- [x] log.js
			- [x] promise.js
			- [ ] thunk.js
			- [ ] wait-service.js
	- pretty-print
		- [ ] worker.js
		- [ ] index.js
    - [x] DevToolsUtils.js
	- [x] assert.js
	- [ ] client.js
	- [x] create-store.js
	- [x] defer.js
	- [ ] dehydrate-state.js
	- [ ] editor.js
	- [ ] fromJS.js
	- [ ] log.js
	- [ ] makeRecord.js
	- [ ] path.js
	- [ ] pause.js
	- [ ] prefs.js
	- [x] test-head.js

- [x] src/feature.js
- [ ] src/main.js

### CODE REVIEW

File: src/utils/DevToolsUtils.js [hunk 1]
```diff
@@ -9,4 +9,4 @@ function executeSoon(fn) {
   setTimeout(fn, 0);
 }
 
-module.exports = { reportException, executeSoon, assert };
+export { reportException, executeSoon, assert };
```
Comment (AnshulMalik) [inline src/utils/DevToolsUtils.js]: We can use default export here?

Comment (jasonLaster) [inline src/utils/DevToolsUtils.js]: It would be nice. @peterschussheim what do you think?

Comment (peterschussheim) [inline src/utils/DevToolsUtils.js]: Yes, good catch! Working on this PR now 💯

File: src/utils/editor/source-documents.js [hunk 2]
```diff
@@ -18,9 +18,4 @@ function clearDocuments() {
   sourceDocs = {};
 }
```
Comment (jasonLaster) [inline src/utils/editor/source-documents.js]: in general we should export each function independently

```js
export function getdocument() {}
```

File: src/utils/redux/middleware/history.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ import type { ThunkArgs } from "../../../actions/types";
  * in logging object. Should only be used for tests, as it collects all
  * action information, which will cause memory bloat.
  */
-exports.history = (log: Object[] = []) => ({
+export const history = (log: Object[] = []) => ({
```
Comment (jasonLaster) [inline src/utils/redux/middleware/history.js]: ooh nice, why is this needed? because it's a declaration?

**Comment by jasonLaster** [comment]
Merging before this gets any bigger :) Thanks Peter!
